File: BackEnd/Domain/Aggregates/TicketStatus.cs
using SharedKernel.Aggregate;

namespace Domain.Aggregates;

public class TicketStatus : AggregateRoot<Guid>
{
    public Guid ProjectId { get; set; }
    public string Name { get; set; }
    public Project Project { get; set; }
    public ICollection<Ticket> Tickets { get; set; }
    
    public static TicketStatus Create(Guid projectId, string name)
    {
        return new TicketStatus
        {
            Id = Guid.CreateVersion7(),
            ProjectId = projectId,
            Name = name,
        };
    }
    public void Update(string name)
    {
        Name = name;
    }
}

File: flutter/lib/domain/aggregates/ticket_status/ticket_status.dart
import 'package:dart_mappable/dart_mappable.dart';
import 'package:enterprise_management/domain/aggregates/ticket/ticket.dart';
import 'package:enterprise_management/shared_kernel/aggregate/aggregate_root.dart';

part 'ticket_status.mapper.dart';
@MappableClass()
class TicketStatus extends AggregateRoot<String> with TicketStatusMappable {
  TicketStatus({required super.id, required this.projectId, required this.name, required this.tickets});

  final String projectId;
  final String name;
  final List<Ticket>? tickets;
}
